### PR TITLE
py-sherpa: Python modeling / fitting package

### PR DESCRIPTION
**Pull request by cdeil**
This pull request contains an initial version of a Portfile for Sherpa, a Python modeling / fitting package.

Do not merge, it's not working yet.

References:
* https://github.com/sherpa/sherpa
* http://cxc.cfa.harvard.edu/contrib/sherpa/
* https://github.com/sherpa/sherpa/releases/tag/4.9.0

The basic part of the `Portfile` should be OK.

But the selection and configuration of the compiler is not, it results in a build error.

https://gist.github.com/cdeil/f22cf4879d93eebb4377b96a7f6ead91
Full log is here: https://guide.macports.org/chunked/reference.portgroup.html

Outside Macports, I can build and install Sherpa like this:
```
F90=/opt/local/bin/gfortran-mp-6 /opt/local/bin/python3.5 setup.py install
```

The compiler stuff that's in the Portfile at the moment is just me guessing around based on what I've seen in other ports (e.g. `py-pymc`, `py-spacepy` or `py-scipy` seem to be using a Fortran compiler). I couldn't find docs or advice about the compiler portgroup!?
https://guide.macports.org/chunked/reference.portgroup.html

Possibly other TODOs for this port are
* to take care of this issue mentioned [here](https://github.com/sherpa/sherpa/tree/master/recipes/conda)
```
# This headers are known to collide with astropy's extensions and would prevent astropy from building
rm -f $PREFIX/include/wcs*.h
```
* to add more tools as dependencies:
https://github.com/sherpa/sherpa#source-build
* to use the Macports fftw instead of the bundled one with Sherpa:
https://github.com/sherpa/sherpa#fftw-library

**Basically - could someone familiar with Fortran compiler in Macports please have a look and make a suggestion what to put in the Portfile, or just take over this pull request?**

Also, any other suggestions are very welcome, e.g. this is the first time I'm using the Github portgroup and I'm not sure if it's done correctly.

Looking at this conda build recipe might be helpful:
https://github.com/sherpa/sherpa/tree/master/recipes/conda
and @olaurino as Sherpa maintainer might be available for questions.

### CODE REVIEW

File: python/py-sherpa/Portfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+long_description    Sherpa is a modeling and fitting application for Python. \
+                    It enables the user to construct complex models from \
+                    simple definitions and fit those models to data, using a \
+                    variety of statistics and optimization methods.
+
+platforms           darwin
+license             GPL3+
+
+homepage            https://github.com/sherpa/sherpa
+github.setup        sherpa sherpa 4.9.0
```
Comment (raimue) [inline python/py-sherpa/Portfile]: Move `github.setup` to the top before name. The homepage is then set automatically and can be removed.

File: python/py-sherpa/Portfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+categories-append   science
+description         Sherpa is a modeling and fitting application for Python
+long_description    Sherpa is a modeling and fitting application for Python. \
+                    It enables the user to construct complex models from \
+                    simple definitions and fit those models to data, using a \
+                    variety of statistics and optimization methods.
+
+platforms           darwin
+license             GPL3+
```
Comment (raimue) [inline python/py-sherpa/Portfile]: Please check with `port lint`:
```
Error: invalid license 'GPL3+': missing hyphen before version
```

The correct license string would be `GPL-3+`.

File: python/py-sherpa/Portfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+PortGroup           compilers 1.0
+name                py-sherpa
+version             1.2
+maintainers         gmail.com:Deil.Christoph openmaintainer
+
+dist_subdir         ${name}/${version}
```
Comment (raimue) [inline python/py-sherpa/Portfile]: Why would this be necessary? The distfile already contains a version number.

File: python/py-sherpa/Portfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+PortGroup           compilers 1.0
+name                py-sherpa
+version             1.2
+maintainers         gmail.com:Deil.Christoph openmaintainer
```
Comment (raimue) [inline python/py-sherpa/Portfile]: You can also add your GitHub username:
```
maintainers         {@cdeil gmail.com:Deil.Christoph} \
                    openmaintainer
```

File: python/py-sherpa/Portfile [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+PortGroup           compilers 1.0
+name                py-sherpa
+version             1.2
```
Comment (raimue) [inline python/py-sherpa/Portfile]: This version number contradicts the `github.setup` line below, which will overwrite it anyway. Move `github.setup` to the top.

File: python/py-sherpa/Portfile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    depends_build-append  port:py${python.version}-setuptools \
+                          port:py${python.version}-numpy
+
+    depends_run-append    port:py${python.version}-numpy \
+                          port:py${python.version}-six \
+                          port:py${python.version}-pytest
+
+    configure.env-append  F90=${prefix}/bin/gfortran-mp-5
+    build.env-append      F90=${prefix}/bin/gfortran-mp-5
```
Comment (raimue) [inline python/py-sherpa/Portfile]: This implies a dependency on `gcc5`, which is not in the Portfile.

**Comment by pmetzger** [comment]
This has been open for quite some time. @cdeil, are you still interested in this?

**Comment by pmetzger** [comment]
@cdeil Are you still working on this, or would you prefer that we close it?

**Comment by pmetzger** [comment]
Okay, so as it stands, this doesn't feel like it's currently an active pull request because it isn't yet in a position to pull and it's been hanging out in the queue for over half a year. If no one speaks up, I'm going to close this in a few days. It can always be reopened with a single click.

**Comment by pmetzger** [comment]
Okay, I'm closing this for now. It can always be reopened.
